app/src/main/admin/components/ModellingGroups/SingleGroup/Details/ViewModellingGroupDetailsPage.tsx

```diff
@@ -7,20 +7,34 @@ import { groupStore } from "../../../../stores/GroupStore";
 import { modellingGroupActions } from "../../../../../shared/actions/ModellingGroupActions";
 import { doNothing } from "../../../../../shared/Helpers";
 import { userStore } from "../../../../stores/UserStore";
+import {IPageWithParent} from "../../../../../shared/models/Breadcrumb";
+import {ViewAllModellingGroupsPage} from "../../List/ViewAllModellingGroupsPage";
 
 export interface ModellingGroupDetailsPageProps {
     groupId: string;
 }
 
 export class ViewModellingGroupDetailsPage extends AdminPageWithHeader<ModellingGroupDetailsPageProps> {
-    componentDidMount() {
-        setTimeout(() => {
-            groupStore.fetchGroups().catch(doNothing).then(() => {
-                modellingGroupActions.setCurrentGroup(this.props.location.params.groupId);
-                groupStore.fetchGroupDetails().catch(doNothing);
-            });
-            userStore.fetchUsers().catch(doNothing);
+    load() {
+        super.load();
+        groupStore.fetchGroups().catch(doNothing).then(() => {
+            modellingGroupActions.setCurrentGroup(this.props.location.params.groupId);
+            groupStore.fetchGroupDetails().catch(doNothing);
         });
+        userStore.fetchUsers().catch(doNothing);
+    }
+
+    name(): string {
+        return "Group details";
```

Comment:
Here it would be more useful to have the group's name

Reply:
This is actually not easy to do, due to the order in which data arrives. I'd like to keep this PR/ticket from getting any more complicated, so I'll raise another ticket for this.